rector.php

```diff
@@ -30,12 +29,11 @@
     ->withPaths([
         __DIR__ . '/.',
     ])
-    ->withImportNames()
+    ->withImportNames(true, true, false, true)
```

Comment:
I suggest to use named arguments, so it is clear what this "true"/"false" is all about. Also rector uses it in the docs:
https://getrector.com/documentation/import-names

Reply:
The TYPO3 Core just decided not to suggest usage of named arguments, so I would stick to that

Reply:
The TYPO3 Core for its API, yes (as they want to be free to rename arguments, but DTOs are fine also in TYPO3 and it is also encouraged in Core to use there named arguments), but this is rector and rector suggests it. Or can you say easily, what the false is about?

Reply:
TYPO3 recommends to assume that argument namesare not API, unless package is clearly declare it as API. If rector really define argument names as API, named arguments are okay (if they really follow semver on that and not break it on patchlevel level).

Package not making a statement about this, should be considered as argument names not beeing API and TYPO3 core **must** not use named arguments to avoid breaking on patchlevel level, and recommends that as a generic thinking.

rector has been heavily breaking on patchlevel level all the time, so I personally would not count that argument names are safe anyway. However, if they suggest it they may add parameter ordering on will - so hard to decide in this specific case.

From generic TYPO3 recommendation and own takedown, named arguments can be used here if rector officially declares it as API (and not only use it in examples). If not, it should be avoided.

My five cents on this aligned to TYPO3 core statements about that topic:

https://docs.typo3.org/m/typo3/reference-coreapi/main/en-us/CodingGuidelines/CodingBestPractices/NamedArguments.html#:~:text=TYPO3%20Core%20may%20leverage%20named,named%2C%20mixtures%20are%20not%20allowed.

Reply:
Rector use named arguments in their examples all over in the docs, for example (yes, that are examples, but as they are "official", I count them as best practise):
- https://getrector.com/documentation/integration-to-new-project
- https://getrector.com/documentation/set-lists
- https://getrector.com/documentation/config-configuration

Rector provides methods with sometimes many, many arguments, so this would not be handable without named arguments (have a look at the `->withPhpSets()` method) - or do you want to have more then 10 "false" before the first relevant "true"?

And, the current versions since 1.0 seem quite stable.

Reply:
I am convinced to use it regarding rector. Done